apps/gallery/test/marionette/lib/gallery.js

```diff
@@ -310,6 +386,45 @@ Gallery.prototype = {
     }.bind(this));
   },
 
+  switchToGalleryFrame: function() {
+    var galleryFrame = this.client.findElement('iframe[src*="' +
+                                          Gallery.ORIGIN + '"]');
+    this.waitFor(galleryFrame);
+    this.client.switchToFrame(galleryFrame);
+  },
+
+  tapFirstThumbnail: function() {
+    this.waitFor(Gallery.Selector.thumbnail);
+    var thumbnails = this.thumbnails;
+    this.waitFor(thumbnails[0]);
+    thumbnails[0].click();
+  },
+
+  selectImage: function() {
+    this.switchToGalleryFrame();
+    this.tapFirstThumbnail();
+  },
+
+  shareImage: function(appName) {
+    this.waitFor(Gallery.Selector.shareButton).tap();
+    this.menuOptionButton(appName).tap();
+  },
+
+  /**
+  * For gallery app open activity, check CSS style background-image
+  * is set with blob URL
+  */
+  hasBackgroundImageBlobURL: function() {
+    var url = 'blob:' + Gallery.ORIGIN;
+    return this.openActivityImage
+               .getAttribute('style').indexOf(url) > -1;
+  },
+
+  switchTo: function() {
+    this.client.switchToFrame();
+    this.client.apps.switchToApp(Gallery.ORIGIN);
+  },
```

Comment:
I don't understand how this function is different from switchToGalleryFrame(), and if I had to maintain this test, I'd just have to experiment with the two of them to figure out when to use each one.  Maybe you can clarify this with comments.